components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -1107,6 +1108,15 @@ func startAPIEndpoint(ctx context.Context, cfg *Config, wg *sync.WaitGroup, serv
 		)
 	}
 
+	grpcMetrics := grpc_prometheus.NewServerMetrics()
+	grpcMetrics.EnableHandlingTimeHistogram(
+		grpc_prometheus.WithHistogramBuckets([]float64{.005, .025, .05, .1, .5, 1, 2.5, 5, 30, 60, 120, 240, 600}),
+	)
+	opts = append(opts,
+		grpc.StreamInterceptor(grpcMetrics.StreamServerInterceptor()),
+		grpc.UnaryInterceptor(grpcMetrics.UnaryServerInterceptor()),
+	)
```

Comment:
If you use the standard `grpc server` options defined in `common-go/grpc` you'll get this for free, and consistent with other servrs.
The alternative is to use `baseserver` for the gRPC server here to get the same benefits.

Reply:
https://gitpod.slack.com/archives/C01KGM9BH54/p1664192086999129?thread_ts=1664192035.420599&cid=C01KGM9BH54

Reply:
> If you use the standard grpc server options defined in common-go/grpc you'll get this for free, and consistent with other servrs.

@easyCZ Could you point me to the code please?

Reply:
is it https://github.com/gitpod-io/gitpod/blob/3f76aaa0c9140dc7489f0430d1dee1510c11ee8d/components/common-go/grpc/grpc.go#L92 ?

Reply:
Yes, indeed. Or you can also use the higher level function [here](https://github.com/gitpod-io/gitpod/blob/3f76aaa0c9140dc7489f0430d1dee1510c11ee8d/components/common-go/grpc/grpc.go#L86)

Reply:
Let me know if you'd like to pair on this. Also fine to not do this if there's a reason not to re-use the common metrics setup.

Reply:
Supervisor has different lifecycle compare to other services, i.e. it belongs to the workspace pod which is ephemeral and it can cause for instance high cardinality issues because of too many workspaces. I think we will most likely reconsider how we collect metrics, i.e. use push gateway. I would prefer to keep it separated for more control for now. If it turned out to be different, i.e. we need to port already solved issues from baseserver, we ca reconsider later.

Reply:
Makes sense. Thanks for the context!